server/task-distributed/src/main/java/org/apache/james/task/eventsourcing/distributed/RabbitMQTerminationSubscriber.java

```diff
@@ -0,0 +1,140 @@
+    private static final String QUEUE_NAME_PREFIX = "terminationSubscriber";
+    private static final String ROUTING_KEY = "terminationSubscriberRoutingKey";
+
+    private final JsonEventSerializer serializer;
+    private final Mono<Connection> connectionMono;
+    private final ReactorRabbitMQChannelPool channelPool;
+    private final String queueName;
+    private WorkQueueProcessor<OutboundMessage> sendQueue;
+    private Flux<Delivery> listener;
+    private Disposable sendQueueHandle;
+
+    @Inject
+    public RabbitMQTerminationSubscriber(SimpleConnectionPool simpleConnectionPool, JsonEventSerializer serializer) {
+        this.serializer = serializer;
+        this.connectionMono = simpleConnectionPool.getResilientConnection();
+        this.channelPool = new ReactorRabbitMQChannelPool(connectionMono, MAX_CHANNELS_NUMBER);
+        this.queueName = QUEUE_NAME_PREFIX + UUID.randomUUID().toString();
+    }
+
+    public void start() {
+        Sender sender = RabbitFlux.createSender(new SenderOptions()
+            .connectionMono(connectionMono)
+            .channelPool(channelPool)
+            .resourceManagementChannelMono(
+                connectionMono.map(Throwing.function(Connection::createChannel)).cache()));
+
+        Receiver receiver = RabbitFlux.createReceiver(new ReceiverOptions().connectionMono(connectionMono));
+        listener = receiver.consumeAutoAck(queueName);
+
+        sender.declareExchange(ExchangeSpecification.exchange(EXCHANGE_NAME)).block();
+        sender.declare(QueueSpecification.queue(queueName).durable(false)).block();
+        sender.bind(BindingSpecification.binding(EXCHANGE_NAME, ROUTING_KEY, queueName)).block();
+        sendQueue = WorkQueueProcessor.create();
+        sendQueueHandle = sender
+            .send(sendQueue)
+            .subscribeOn(Schedulers.elastic())
+            .subscribe();
+    }
+
+    @Override
+    public void addEvent(Event event) {
+        try {
+            byte[] payload = serializer.serialize(event).getBytes(StandardCharsets.UTF_8);
+            AMQP.BasicProperties basicProperties = new AMQP.BasicProperties.Builder().build();
+            OutboundMessage message = new OutboundMessage(EXCHANGE_NAME, ROUTING_KEY, basicProperties, payload);
+            sendQueue.onNext(message);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Stream<Event> listenEvents() {
+        return listener
+            .concatMap(this::toEvent)
+            .share()
+            .toStream();
+    }
+
+    private Mono<Event> toEvent(Delivery delivery) {
+        String message = new String(delivery.getBody(), StandardCharsets.UTF_8);
+        try {
+            Event event = serializer.deserialize(message);
+            return Mono.just(event);
+        } catch (Exception e) {
+            LOGGER.error("Unable to deserialize '" + message + "'", e);
```

Comment:
Use placeholder for the sake of readability

Reply:
I would not be able to pass exception then, right?

Reply:
No you will